framework/src/binding/store/primitive.rs

```diff
@@ -65,144 +63,157 @@ impl<S: ServiceState> DefaultStoreUint64<S> {
+        let sv = self.inner_get();
 
         match val.overflowing_add(sv) {
-            (sum, false) => self.set_(sum),
-            _ => Err(StoreError::Overflow.into()),
+            (sum, false) => {
+                self.inner_set(sum);
+                false
+            }
+            _ => true,
         }
     }
 
     // Self minus val
     // And set the result back to self
-    fn sub_(&mut self, val: u64) -> ProtocolResult<()> {
-        let sv = self.get_()?;
+    fn inner_sub(&mut self, val: u64) -> bool {
+        let sv = self.inner_get();
 
         if sv >= val {
-            self.set_(sv - val)
+            self.inner_set(sv - val);
+            false
         } else {
-            Err(StoreError::Overflow.into())
+            true
         }
     }
 
     // Multiply val with self
     // And set the result back to self
-    fn mul_(&mut self, val: u64) -> ProtocolResult<()> {
-        let sv = self.get_()?;
+    fn inner_mul(&mut self, val: u64) -> bool {
+        let sv = self.inner_get();
 
         match val.overflowing_mul(sv) {
-            (mul, false) => self.set_(mul),
-            _ => Err(StoreError::Overflow.into()),
+            (mul, false) => {
+                self.inner_set(mul);
+                false
+            }
+            _ => true,
         }
     }
 
     // Power of self
     // And set the result back to self
-    fn pow_(&mut self, val: u32) -> ProtocolResult<()> {
-        let sv = self.get_()?;
+    fn inner_pow(&mut self, val: u32) -> bool {
+        let sv = self.inner_get();
 
         match sv.overflowing_pow(val) {
-            (pow, false) => self.set_(pow),
-            _ => Err(StoreError::Overflow.into()),
+            (pow, false) => {
+                self.inner_set(pow);
+                false
+            }
+            _ => true,
         }
     }
 
     // Self divided by val
     // And set the result back to self
-    fn div_(&mut self, val: u64) -> ProtocolResult<()> {
-        let sv = self.get_()?;
+    fn inner_div(&mut self, val: u64) -> bool {
+        let sv = self.inner_get();
 
         if let 0 = val {
-            Err(StoreError::Overflow.into())
+            true
         } else {
-            self.set_(sv / val)
+            self.inner_set(sv / val);
+            false
         }
     }
 
     // Remainder of self
     // And set the result back to self
-    fn rem_(&mut self, val: u64) -> ProtocolResult<()> {
-        let sv = self.get_()?;
+    fn inner_rem(&mut self, val: u64) -> bool {
+        let sv = self.inner_get();
 
         if let 0 = val {
-            Err(StoreError::Overflow.into())
+            true
         } else {
-            self.set_(sv % val)
+            self.inner_set(sv % val);
+            false
         }
     }
 }
 
 impl<S: ServiceState> StoreUint64 for DefaultStoreUint64<S> {
     fn get(&self) -> u64 {
-        self.get_()
-            .unwrap_or_else(|e| panic!("StoreUint64 get failed: {}", e))
+        self.inner_get()
```

Comment:
I think we can remove `fn inner_xx` method, and move the logic to `fn xx` trait method.
The reason of `inner_xx` existed is historical.

Reply:
Agreed, but refactoring may not be included here because the focus of this PR is to solve the overflow.